dali/python/nvidia/dali/plugin/jax/iterator.py

```diff
@@ -317,17 +319,25 @@ def create_iterator(*args, **wrapper_kwargs):
                             "running in multiprocess mode. Use `sharding` argument instead."
                         )
 
+                if checkpoints and len(checkpoints) != len(devices_to_use):
+                    raise RuntimeError(
+                        "The number of checkpoints provided should match the number of devices "
+                        "to use."
```

Comment:
Could you maybe show how those numbers differ?

Reply:
Good idea, fixed